packages/ckeditor5-engine/src/view/observer/selectionobserver.ts

```diff
@@ -232,6 +232,10 @@ export default class SelectionObserver extends Observer {
 
 		const domSelection = domDocument.defaultView!.getSelection()!;
 
+		if ( isGeckoRestrictedDomSelection( domSelection ) ) {
+			return;
+		}
```

Comment:
I'm not sure if this code should be checked right here. I'm wondering if it should not be verified in `DomConverter#domSelectionToView()` and selection without ranges should be returned.

Reply:
That would be after `this.mutationObserver.flush()`. Do you think this would be OK?

Reply:
I think so. What we miss in this PR is changing the hasDomSelection flag. The selection is no longer in editable so we should handle it as it would have no ranges.